src/rust/pants_ng/options/src/options.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2025 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use crate::config::ConfigFinder;
+use crate::config::InterpolationMap;
+use crate::pants_invocation::PantsInvocation;
+use options::OptionParser;
+use options::Source;
+use options::env::EnvReader;
+use options::fromfile::FromfileExpander;
+use options::pants_ng_flags::PantsNgFlagsReader;
+use options::{BuildRoot, Env, OptionsSource};
+use std::collections::BTreeMap;
+use std::collections::HashSet;
+use std::path::Path;
+use std::sync::Arc;
+
+pub struct OptionsReader {
+    #[allow(dead_code)]
+    options_parser: OptionParser,
+}
+
+pub struct Options {
+    pants_invocation: PantsInvocation,
+    env: Env,
+    config_finder: ConfigFinder,
+    context: HashSet<String>,
+    fromfile_expander: FromfileExpander,
+    include_derivation: bool,
+}
+
+impl Options {
+    pub fn new(
+        pants_invocation: PantsInvocation,
+        env: Env,
+        buildroot: Option<BuildRoot>,
+        seed_values: InterpolationMap,
+        context: Vec<String>,
+        include_derivation: bool,
+    ) -> Result<Self, String> {
+        let buildroot = buildroot.unwrap_or(BuildRoot::find()?);
```

Comment:
Maybe use `.unwrap_or_else` to avoid always searching for a buildroot even when one is provided? (I.e., `BuildRoot::find()` is always executed as written here.)

Reply:
Can't use `unwrap_or_else` because the closure would return a Result, but have changed to use `if let ... else ...` to the same effect, so the buildroot is only computed as needed.